store/storetest/team_store.go

```diff
@@ -678,6 +681,89 @@ func testGetAllPrivateTeamPageListing(t *testing.T, ss store.Store) {
 	}
 }
 
+func testGetAllPublicTeamPageListing(t *testing.T, ss store.Store) {
+	o1 := model.Team{}
+	o1.DisplayName = "DisplayName"
+	o1.Name = "z-z-z" + model.NewId() + "b"
+	o1.Email = MakeEmail()
+	o1.Type = model.TEAM_OPEN
+	o1.AllowOpenInvite = true
+	_, err := ss.Team().Save(&o1)
+	require.Nil(t, err)
+
+	o2 := model.Team{}
+	o2.DisplayName = "DisplayName"
+	o2.Name = "zz" + model.NewId() + "b"
+	o2.Email = MakeEmail()
+	o2.Type = model.TEAM_OPEN
+	o2.AllowOpenInvite = false
+	_, err = ss.Team().Save(&o2)
+	require.Nil(t, err)
+
+	o3 := model.Team{}
+	o3.DisplayName = "DisplayName"
+	o3.Name = "z-z-z" + model.NewId() + "b"
+	o3.Email = MakeEmail()
+	o3.Type = model.TEAM_INVITE
+	o3.AllowOpenInvite = true
+	_, err = ss.Team().Save(&o3)
+	require.Nil(t, err)
+
+	o4 := model.Team{}
+	o4.DisplayName = "DisplayName"
+	o4.Name = "zz" + model.NewId() + "b"
+	o4.Email = MakeEmail()
+	o4.Type = model.TEAM_INVITE
+	o4.AllowOpenInvite = false
+	_, err = ss.Team().Save(&o4)
+	require.Nil(t, err)
+
+	if teams, listErr := ss.Team().GetAllPublicTeamPageListing(0, 10); listErr != nil {
+		t.Fatal(listErr)
+	} else {
+		for _, team := range teams {
+			require.Equal(t, team.AllowOpenInvite, true, "should have returned team with AllowOpenInvite as true")
+		}
+
+		if len(teams) > 10 {
+			t.Fatal("should have returned max of 10 teams")
+		}
+	}
+
+	o5 := model.Team{}
+	o5.DisplayName = "DisplayName"
+	o5.Name = "z-z-z" + model.NewId() + "b"
+	o5.Email = MakeEmail()
+	o5.Type = model.TEAM_OPEN
+	o5.AllowOpenInvite = true
+	_, err = ss.Team().Save(&o5)
+	require.Nil(t, err)
+
+	if teams, listErr := ss.Team().GetAllPublicTeamPageListing(0, 4); listErr != nil {
```

Comment:
Could this just get us the 4 teams created above by fluke, even if the functionality being tested itself is failing? This also feels a bit brittle. If we are confident there are only 5 teams in the DB then getting all 5 should result in only the 4 public ones being returned.

Reply:
@grundleborg, thanks for the thorough review and catching these.  I copied and adjusted the test from another test, but am going to adjust this one so that I'll be testing for the exact teams that are added for the tests.